homeassistant/components/sensor/thermoworks_smoke.py

```diff
@@ -0,0 +1,151 @@
+    CONF_MONITORED_VARIABLES, CONF_EXCLUDE
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['thermoworks_smoke==0.1.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+PROBE_1 = 'probe1'
+PROBE_2 = 'probe2'
+PROBE_1_MIN = 'probe1Min'
+PROBE_1_MAX = 'probe1Max'
+PROBE_2_MIN = 'probe2Min'
+PROBE_2_MAX = 'probe2Max'
+
+SENSOR_TYPES = {
+    PROBE_1: 'Probe 1',
+    PROBE_2: 'Probe 2',
+    PROBE_1_MIN: 'Probe 1 Min',
+    PROBE_1_MAX: 'Probe 1 Max',
+    PROBE_2_MIN: 'Probe 2 Min',
+    PROBE_2_MAX: 'Probe 2 Max',
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=[PROBE_1, PROBE_2]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_EXCLUDE, default=[]):
+        vol.All(cv.ensure_list),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the thermoworks sensor."""
+    import thermoworks_smoke
+
+    email = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+    monitored_variables = config.get(CONF_MONITORED_VARIABLES)
+    excluded = config.get(CONF_EXCLUDE)
+
+    mgr = thermoworks_smoke.initialize_app(email, password, True, excluded)
+
+    # list of sensor devices
+    dev = []
+
+    # get list of registered devices
+    for serial in mgr.serials():
+        for variable in monitored_variables:
+            dev.append(ThermoworksSmokeSensor(variable, serial, mgr))
+
+    add_devices(dev, True)
+
+
+class ThermoworksSmokeSensor(Entity):
+    """Implementation of a thermoworks smoke sensor."""
+
+    def __init__(self, sensor_type, serial, mgr):
+        """Initialize the sensor."""
+        self._name = mgr.name(serial) + ' ' + SENSOR_TYPES[sensor_type]
+        self.type = sensor_type
+        self._state = None
+        self._attributes = {}
+        self._unit_of_measurement = TEMP_FAHRENHEIT
+        self.serial = serial
+        self.mgr = mgr
+        self.update_unit()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this sensor."""
+        return self._unit_of_measurement
+
+    def update_unit(self):
+        """Set the units from the data."""
+        if PROBE_2 in self.type:
+            self._unit_of_measurement = self.mgr.units(self.serial, PROBE_2)
+        else:
+            self._unit_of_measurement = self.mgr.units(self.serial, PROBE_1)
+
+    def update(self):
+        """Get the monitored data from firebase."""
+        try:
+            values = self.mgr.data(self.serial)
```

Comment:
The data should be stored in a data object to keep the amount of requests low. Instead of 6 requests will there only be one to get the values for possible sensors.

Reply:
The thermoworks python project does all the caching. Calls to data are cheap.